cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -368,11 +369,18 @@ public void performAction(HttpRequest request, HttpResponder responder,
     }
 
     if (!isValidAction(action)) {
-      responder.sendStatus(HttpResponseStatus.METHOD_NOT_ALLOWED);
+      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
```

Comment:
I think now we throw exception for error response. E.g. `throw new NotFoundException(...)` in this case. There is a error handler that will turn known exception into corresponding response code and respond.